stablehlo/tests/ops_stablehlo_quantized.mlir

```diff
@@ -23,10 +24,50 @@ func.func @ops_per_axis_quantization(
   %tuple = "stablehlo.tuple"(%arg1, %arg1) : (tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:0, {0.1:-30}>>, tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:0, {0.1:-30}>>) -> tuple<tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:0, {0.1:-30}>>, tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:0, {0.1:-30}>>>
   %uniform_dequantize = "stablehlo.uniform_dequantize" (%arg0) : (tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:2, {0.1:-30, 0.5:-20}>>) -> tensor<1x2x2xf32>
   %uniform_quantize = "stablehlo.uniform_quantize" (%arg0) : (tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:2, {0.1:-30, 0.5:-20}>>) -> tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:2, {0.1:-30, 0.5:-20}>>
-  func.return
-}
+  func.return %bitcast_convert, %broadcast_in_dim_1, %broadcast_in_dim_2, %custom_call, %dynamic_broadcast_in_dim, %get_dimension_size, %outfeed, %reshape, %send, %transpose, %tuple, %uniform_dequantize, %uniform_quantize : tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:2, {0.1:-30, 0.5:-20}>>, tensor<1x2x3x2x!quant.uniform<i8<-128:127>:f32:3, {0.1:-30, 0.5:-20}>>, tensor<2x2x2x!quant.uniform<i8<-128:127>:f32:0, {0.1:-30, 0.1:-30}>>, tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:2, {0.1:-30, 0.5:-20}>>, tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:2, {0.1:-30, 0.5:-20}>>, tensor<i32>, !stablehlo.token, tensor<2x2x!quant.uniform<i8<-128:127>:f32:1, {0.1:-30, 0.5:-20}>>, !stablehlo.token, tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:1, {0.1:-30, 0.5:-20}>>, tuple<tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:0, {0.1:-30}>>, tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:0, {0.1:-30}>>>, tensor<1x2x2xf32>, tensor<1x2x2x!quant.uniform<i8<-128:127>:f32:2, {0.1:-30, 0.5:-20}>>
+}
+
+// CHECK-QDQ-NOT: stablehlo.uniform_dequantize
```

Comment:
These first two tests are monolithic and will likely be more difficult to maintain (and certainly more difficult to understand what is going on in the check lines.

I see some per-op tests below, what is the coverage of these? Can we rely on QDQ testing just using those per-op tests?

Reply:
The bulkiest function in this file `ops_per_tensor_quantization` covers most of the elementwise quantized operations which which are supposed to be affected by this pass. In comparison, the following per-op tests covers very few operations  (batch_norm_*, max/min, round*, compare) actually need dq-op-q treatment.

I have added all the checks with context, hopefully this would help.